hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/ReplicatedMapEntryView.java

```diff
@@ -33,21 +34,31 @@
 
     private static final int NOT_AVAILABLE = -1;
 
-    private K key;
-    private V value;
+    private Object key;
+    private Object value;
     private long creationTime;
     private long hits;
     private long lastAccessTime;
     private long lastUpdateTime;
     private long ttl;
     private long maxIdle;
 
+    private SerializationService serializationService;
+
     public ReplicatedMapEntryView() {
     }
 
+    public ReplicatedMapEntryView(SerializationService serializationService) {
+        this.serializationService = serializationService;
+    }
+
     @Override
     public K getKey() {
-        return key;
+        // `null` `serializationService` means, use raw type without any conversion
```

Comment:
minor: markdown markup in comment is not respected unfortunately... (I also do this all the time :-) )

Reply:
copy&paste error. fixed.